pkg/reconciler/testing/functional.go

```diff
@@ -1,1118 +0,0 @@
-		pa.Spec.ContainerConcurrency = cc
-	}
-}
-
-func withAnnotationValue(key, value string) PodAutoscalerOption {
-	return func(pa *autoscalingv1alpha1.PodAutoscaler) {
-		if pa.Annotations == nil {
-			pa.Annotations = make(map[string]string)
-		}
-		pa.Annotations[key] = value
-	}
-}
-
-// WithTargetAnnotation returns a PodAutoscalerOption which sets
-// the PodAutoscaler autoscaling.knative.dev/target annotation to the
-// provided value.
-func WithTargetAnnotation(target string) PodAutoscalerOption {
-	return withAnnotationValue(autoscaling.TargetAnnotationKey, target)
-}
-
-// WithWindowAnnotation returns a PodAutoScalerOption which sets
-// the PodAutoscaler autoscaling.knative.dev/window annotation to the
-// provided value.
-func WithWindowAnnotation(window string) PodAutoscalerOption {
-	return withAnnotationValue(autoscaling.WindowAnnotationKey, window)
-}
-
-// WithPanicThresholdPercentageAnnotation returns a PodAutoscalerOption
-// which sets the PodAutoscaler
-// autoscaling.knative.dev/targetPanicPercentage annotation to the
-// provided value.
-func WithPanicThresholdPercentageAnnotation(percentage string) PodAutoscalerOption {
-	return withAnnotationValue(autoscaling.PanicThresholdPercentageAnnotationKey, percentage)
-}
-
-// WithWindowPanicPercentageAnnotation retturn a PodAutoscalerOption
-// which set the PodAutoscaler
-// autoscaling.knative.dev/windowPanicPercentage annotation to the
-// provided value.
-func WithPanicWindowPercentageAnnotation(percentage string) PodAutoscalerOption {
-	return withAnnotationValue(autoscaling.PanicWindowPercentageAnnotationKey, percentage)
-}
-
-// WithMetricAnnotation adds a metric annotation to the PA.
-func WithMetricAnnotation(metric string) PodAutoscalerOption {
-	return withAnnotationValue(autoscaling.MetricAnnotationKey, metric)
-}
-
-// K8sServiceOption enables further configuration of the Kubernetes Service.
-type K8sServiceOption func(*corev1.Service)
-
-// OverrideServiceName changes the name of the Kubernetes Service.
-func OverrideServiceName(name string) K8sServiceOption {
-	return func(svc *corev1.Service) {
-		svc.Name = name
-	}
-}
-
-// MutateK8sService changes the service in a way that must be reconciled.
-func MutateK8sService(svc *corev1.Service) {
-	// An effective hammer ;-P
-	svc.Spec = corev1.ServiceSpec{}
-}
-
-// WithClusterIP assigns a ClusterIP to the K8s Service.
-func WithClusterIP(ip string) K8sServiceOption {
-	return func(svc *corev1.Service) {
-		svc.Spec.ClusterIP = ip
-	}
-}
-
-// WithExternalName gives external name to the K8s Service.
-func WithExternalName(name string) K8sServiceOption {
-	return func(svc *corev1.Service) {
-		svc.Spec.ExternalName = name
-	}
-}
-
-// WithK8sSvcOwnersRemoved clears the owner references of this Service.
-func WithK8sSvcOwnersRemoved(svc *corev1.Service) {
-	svc.OwnerReferences = nil
-}
-
-// EndpointsOption enables further configuration of the Kubernetes Endpoints.
-type EndpointsOption func(*corev1.Endpoints)
```

Comment:
Those I am pretty sure are independent of our API versioning and probably should stay where they are.

Reply:
They don't make sense under `pkg/reconciler`, but I can move them up a level into `pkg/testing` so they are unversioned.

Reply:
Done.